scripts/g.extension/testsuite/test_addons_download.py

```diff
@@ -0,0 +1,164 @@
+           License (>=v2). Read the file COPYING that comes with GRASS
+           for details.
+"""
+
+import os
+from importlib.machinery import SourceFileLoader
+from distutils.spawn import find_executable
+
+from grass.gunittest.case import TestCase
+from grass.gunittest.main import test
+from grass.gunittest.utils import silent_rmtree
+
+
+class TestModuleDownloadFromDifferentSources(TestCase):
+
+    # MS Windows install function requires absolute paths
+    install_prefix = os.path.abspath("gextension_test_install_path")
+
+    files = [
+        os.path.join(install_prefix, "scripts", "r.example.plus"),
+        os.path.join(install_prefix, "docs", "html", "r.example.plus.html"),
+    ]
+
+    def setUp(self):
+        """Make sure we are not dealing with some old files"""
+        if os.path.exists(self.install_prefix):
+            files = os.listdir(self.install_prefix)
+            if files:
+                RuntimeError(
+                    "Install prefix path '{}' contains files {}".format(
+                        self.install_prefix, files
+                    )
+                )
+
+    def tearDown(self):
+        """Remove created files"""
+        silent_rmtree(self.install_prefix)
+
+    def test_github_install(self):
+        """Test installing extension from github"""
+
+        self.assertModule(
+            "g.extension",
+            extension="r.example.plus",
+            url="https://github.com/wenzeslaus/r.example.plus",
+            prefix=self.install_prefix,
+        )
+
+        # Modules with non-standard branch would be good for testing...
+        self.assertModule(
+            "g.extension",
+            extension="r.example.plus",
+            url="https://github.com/wenzeslaus/r.example.plus",
+            prefix=self.install_prefix,
+        )
+
+        for file in self.files:
+            self.assertFileExists(file)
+
+    def test_gitlab_install(self):
+        """Test installing extension from gitlab"""
+        self.assertModule(
+            "g.extension",
+            extension="r.example.plus",
+            url="https://gitlab.com/vpetras/r.example.plus",
+            prefix=self.install_prefix,
+        )
+
+        for file in self.files:
+            self.assertFileExists(file)
+
+    def test_bitbucket_install(self):
+        """Test installing extension from bitbucket"""
+        d_rast_files = [
+            os.path.join(self.install_prefix, "scripts", "r.sim.stats"),
+            os.path.join(self.install_prefix, "docs", "html", "r.sim.stats.html"),
+        ]
+        self.assertModule(
+            "g.extension",
+            extension="r.sim.stats",
+            url="https://bitbucket.org/lrntct/r.sim.stats",
+            prefix=self.install_prefix,
+        )
+
+        for file in d_rast_files:
+            self.assertFileExists(file)
+
+    def test_github_install_official(self):
+        """Test installing C-extension from official addons repository"""
+        r_gdd_files = [
```

Comment:
```suggestion
        files = [
```

Not wrong unlike the other case, but the code is short thus the variable clear. Confusion with self.files is possible, but easily discovered.